### ISSUE TITLE
Incorrect handling of negative zero (TOML 1.1)

### ISSUE THREAD

**Issue by innermatrix** [squirrelchat/smol-toml#30]
```
import toml from 'smol-toml'
toml.parse('a=-0')
```

Expected: `{ a: -0 }`
Observed: `{ a: 0 }`

(The current behavior is correct in TOML 1.0 and incorrect in TOML 1.1; the TOML 1.1 behavior would also be correct in TOML 1.0)

**Comment by cyyynthia** [comment]
TOML 1.1 does not change the semantic of -0 and +0 being equal to an unprefixed 0. "-0" does not exist as a distinct integer value and is only a concept that exists in floating point numbers, which smol preserves correctly (-0.0 != 0.0).

The library is lossless with regards to the value during deserialization (modulo integer vs float distinction; 1.0 == 1).

**Comment by innermatrix** [comment]
Oh fun I'll try to figure out what else in my toolchain mislead me. Thanks!

**Comment by innermatrix** [comment]
tldr a different encoder was serializing JSON `-0` as TOML `-0` (integer) (instead of TOML `-0.0`) and then `smol-toml` was parsing that as JSON `0`, at which point my round-tripping unit tests failed. Then I misread the spec and incorrectly concluded that TOML `-0` should be parsed as JSON `-0`. Thanks for correcting me!
